core/auth/auth.go

```diff
@@ -149,16 +150,23 @@ func NewAuth(serviceID string, host string, authPrivKey *rsa.PrivateKey, storage
 		return nil, errors.WrapErrorAction(logutils.ActionSave, "reg", nil, err)
 	}
 
-	dataLoader := NewLocalDataLoader(storage)
+	authService := authservice.AuthService{
+		ServiceID:   serviceID,
+		ServiceHost: host,
+		FirstParty:  true,
+	}
+
+	serviceRegLoader := NewLocalServiceRegLoader(storage)
 
-	authService, err := authservice.NewAuthService(serviceID, host, dataLoader)
+	// Instantiate a ServiceRegManager to manage the service registration data loaded by serviceRegLoader
+	serviceRegManager, err := authservice.NewServiceRegManager(&authService, serviceRegLoader)
 	if err != nil {
-		return nil, errors.WrapErrorAction(logutils.ActionInitialize, "auth service", nil, err)
+		log.Fatalf("Error initializing service registration manager: %v", err)
```

Comment:
Is there a reason we need to use the `log` package instead of `auth.logger` to do this?